### ISSUE TITLE
Shows both wrong diagnostic and correct symbol tooltip

### ISSUE THREAD

**Issue by abyss7** [isundaylee/atom-ide-ccls#1]
It may look like this:
![screenshot_20181109_140759](https://user-images.githubusercontent.com/5627721/48259708-d568b000-e429-11e8-8b6d-79427e361070.png)

For me it looks like a bug, since both diagnostic and symbol description (even with comment) are provided via `ide-ccls`, am I right?

**Comment by isundaylee** [comment]
Hmm yeah that does look a bit weird. Both diagnostic and symbol info should be coming from ccls. Thanks for reporting this! Is this a public codebase that I can check out?

**Comment by abyss7** [comment]
Yes, it's https://github.com/yandex/clickhouse
Trying on Fedora 29.
CMake command is: 

> cmake . -Bbuild -DMAKE_STATIC_LIBRARIES=OFF -DCMAKE_BUILD_TYPE=Release -DENABLE_EMBEDDED_COMPILER=OFF -DCMAKE_EXPORT_COMPILE_COMMANDS=ON

**Comment by isundaylee** [comment]
Thanks for the info! Which file were you looking at? I tried grep-ing but there are multiple matches.

**Comment by abyss7** [comment]
Almost every file contained some errors - doesn't really matter which one.

**Comment by isundaylee** [comment]
Interesting, I took a look at `dbms/src/Storages/System/StorageSystemPartsBase.h`, but I'm only seeing the diagnostic messages, but not the symbol info (see screenshot below). Is it possible that the symbol information is coming from another plugin? 

![screen shot 2018-11-09 at 10 01 21 am](https://user-images.githubusercontent.com/1572861/48269861-9412d900-e406-11e8-9b33-a38b7a987af8.png)

**Comment by isundaylee** [comment]
BTW I noticed that cmake generated the `compile_commands.json` file in the `build` subdirectory. I think ccls looks for the file in the project root directory. A lot of the diagnostic messages went away after running the following in the project root to symlink the compilation database:

```bash
ln -s build/compile_commmands.json .
```

**Comment by abyss7** [comment]
That is possible - I will double-check later. But anyway the diagnostics seem wrong. I cross-checked with ide-cquery on the same compilation database - everything works fine.

**Comment by abyss7** [comment]
> BTW I noticed that cmake generated the `compile_commands.json` file in the `build` subdirectory. I think ccls looks for the file in the project root directory. A lot of the diagnostic messages went away after running the following in the project root to symlink the compilation database:
> 
> ```shell
> ln -s build/compile_commmands.json .
> ```

Yes, I did the same trick - but with no success.

**Comment by isundaylee** [comment]
Yeah I'm still getting some weird diagnostic messages. Let me look into that a bit further.

**Comment by isundaylee** [comment]
Actually I'm able to repro your original issue in a different file. Investigating that now.
